code/game/objects/items/reagent_containers/pill.dm

```diff
@@ -88,15 +88,25 @@ var/global/list/randomized_pill_icons
 		return
 
 	if(target.is_refillable())
-		if(target.is_drainable() && !target.reagents.total_volume)
-			to_chat(user, "<span class='warning'>[target] is empty! There's nothing to dissolve [src] in.</span>")
-			return
-
 		if(target.reagents.holder_full())
 			to_chat(user, "<span class='warning'>[target] is full.</span>")
 			return
 
-		to_chat(user, "\blue You dissolve the pill in [target].</span>")
+		var/obj/item/reagent_container/R = null
+		var/liquidate = null
+		if(istype(target,/obj/item/reagent_container))
+			R = target
+			if(R.liquifier)
```

Comment:
Why not use R.liquifier instead of setting a new variable for it? It seems a little unnecessary especially since it's already pretty short and means the same thing really.